craft_parts/plugins/gradle_plugin.py

```diff
@@ -0,0 +1,165 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2015-2023,2024,2025 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The gradle plugin."""
+
+import os
+from pathlib import Path
+from typing import Literal, cast
+from urllib.parse import urlparse
+
+from overrides import override
+from pydantic import model_validator
+from typing_extensions import Self
+
+from . import validator
+from .java_plugin import JavaPlugin
+from .properties import PluginProperties
+
+GRADLE_PLUGIN_VALIDATOR_FILE = Path(__file__).parent / "gradle_plugin_validator.py"
+
+
+class GradlePluginProperties(PluginProperties, frozen=True):
+    """The part properties used by the gradle plugin."""
+
+    plugin: Literal["gradle"] = "gradle"
+
+    gradle_init_script: str = ""
+    gradle_init_script_parameters: list[str] = []
+    gradle_parameters: list[str] = []
+    gradle_task: str = "build"
+
+    # part properties required by the plugin
+    source: str  # pyright: ignore[reportGeneralTypeIssues]
```

Comment:
What is the specific problem here?

Reply:
I'm getting the following error from Pywright:
```
/home/canonical/craft-parts/craft_parts/plugins/gradle_plugin.py:46:5 - error: "source" overrides a field of the same name but is missing a default value (reportGeneralTypeIssues)
```

This is because `PluginProperties` parent class defines source as `source: str | None = None` and we're overriding it. Other plugins also have the same type ignores to help PyWright :)

Reply:
Ah ok I see